pkg/suggestion/bayesianoptimization/src/algorithm_manager.py

```diff
@@ -1,39 +1,31 @@
 """ module for algorithm manager """
-
 import numpy as np
 
 from pkg.api.python import api_pb2
-import logging
-from logging import getLogger, StreamHandler, INFO, DEBUG
+
+from .utils import get_logger
+
 
 def deal_with_discrete(feasible_values, current_value):
     """ function to embed the current values to the feasible discrete space"""
     diff = np.subtract(feasible_values, current_value)
     diff = np.absolute(diff)
     return feasible_values[np.argmin(diff)]
 
+
 def deal_with_categorical(feasible_values, one_hot_values):
     """ function to do the one hot encoding of the categorical values """
-    #index = np.argmax(one_hot_values)
-    index = one_hot_values.argmax()
+    index = np.argmax(one_hot_values)
+    #index = one_hot_values.argmax()
```

Comment:
Why did you change this?

Reply:
The two lines perform the same task. However, the np.argmax function works on any array_like object (e.g. list, tuple, or ndarray), while the ndarray.argmax() method only works on ndarrays. I switched for two reasons, i) the old version raised an exception because it was being passed a list at some point, ii) the function approach is more flexible and requires less type casts to force array_like variables into ndarrays.